In [171]:
from scipy.linalg import svd, diagsvd
import matplotlib.image as mpimg
import numpy as np

m1 = np.array([[3,4,3,4],[5,2,2,3],[2,1,5,3],[5,3,4,1]])

m2 = np.array([[3,4,np.random.randint(0,5),4],[5,2,2,3],[2,1,5,3],[5,3,4,1]])

u, s, vt = svd(m2)

s[-3] *= 0

sigma = diagsvd(s, m1.shape[0], m1.shape[1])

B = u @ sigma @ vt 
B

array([[2.64943177, 2.58225427, 2.77954782, 2.92946248],
       [4.92861588, 1.71131288, 2.56598276, 2.78201282],
       [2.27291668, 2.10371224, 2.83612753, 3.83341134],
       [5.12874774, 3.5206734 , 2.97919875, 1.39315964]])

$$
A = \begin{bmatrix}
3 & 4 & 3 & 4 \\
5 & 2 & 2 & 3 \\
2 & 1 & 5 & 3 \\
5 & 3 & 4 & 1
\end{bmatrix}
$$

$$
A = \begin{bmatrix}
2.649 & 2.582 & 2.779 & 2.929 \\
4.928 & 1.711 & 2.565 & 2.782 \\
2.272, & 2.103 & 2.836 & 3.833 \\
5.128 & 3.520 & 3.520 & 1.393
\end{bmatrix}
$$